@stellar/typescript-wallet-sdk/src/walletSdk/Server/index.ts

```diff
@@ -58,3 +66,23 @@ export const signChallengeTransaction = async ({
     networkPassphrase,
   };
 };
+
+/**
+ * Helper method for parsing a JSON string into an AnchorTransaction.
+ * @param {string} transaction - The json string of an anchor transaction.
+ * @returns {AnchorTransaction} The transaction object.
+ */
+export const parseAnchorTransaction = (
+  transaction: string,
+): AnchorTransaction => {
+  const parsed = JSON.parse(transaction);
```

Comment:
I believe `JSON.parse()` alone can throw, should we wrap it with `try catch` just in case we receive something not well formatted from the server?

Reply:
And throw a different error in this case, something like `TransactionIsNotValidJSONError()` or similar.

Too conservative?

Reply:
ya I was on the fence if the try-catch should be handled here or by the function using this method.

But I think you've nudged me over the edge :)
https://github.com/stellar/typescript-wallet-sdk/pull/111/commits/4684622ba08d895371fce0535baabbe7d3d236e4